Lib/test/test_complex.py

```diff
@@ -527,46 +501,103 @@ class complex2(complex):
             def __complex__(self):
                 return None
 
-        self.assertEqual(complex(complex0(1j)), 42j)
+        check(complex(complex0(1j)), 0.0, 42.0)
         with self.assertWarns(DeprecationWarning):
-            self.assertEqual(complex(complex1(1j)), 2j)
+            check(complex(complex1(1j)), 0.0, 2.0)
```

Comment:
@serhiy-storchaka, maybe it's time to drop this?  Ditto for float's.

Returning non-complex(float) values was deprecated in v3.7.

Reply:
This is a different issue.